modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/StandaloneASDiscovery.java

```diff
@@ -82,17 +82,28 @@ protected String getLogFileName() {
     @Override
     protected String buildDefaultResourceName(HostPort hostPort, HostPort managementHostPort,
         JBossProductType productType, String serverName) {
+        return super.buildDefaultResourceName(hostPort, managementHostPort, productType, serverName);
+    }
+
+    @Override
+    String buildDefaultResourceName(HostPort hostPort, HostPort managementHostPort,
+        JBossProduct product, String serverName) {
         if (serverName != null && !serverName.trim().isEmpty()) {
-            return String.format("%s (%s %s:%d)", productType.SHORT_NAME, serverName, managementHostPort.host,
+            return String.format("%s (%s %s:%d)", product.SHORT_NAME, serverName, managementHostPort.host,
                 managementHostPort.port);
         }
 
-        return String.format("%s (%s:%d)", productType.SHORT_NAME, managementHostPort.host, managementHostPort.port);
+        return String.format("%s (%s:%d)", product.SHORT_NAME, managementHostPort.host, managementHostPort.port);
     }
 
     @Override
     protected String buildDefaultResourceDescription(HostPort hostPort, JBossProductType productType) {
-        return String.format("Standalone %s server", productType.FULL_NAME);
+        return super.buildDefaultResourceDescription(hostPort, productType);
+    }
+
+    @Override
+    public String buildDefaultResourceDescription(HostPort hostPort, JBossProduct product) {
```

Comment:
I intentionally made this public so this 2 methods could be called from ResourceDiscoveryCallback implementations in other plugins. But at the same time I provided abstract callback implementation (in this PR), which should be enough for plugins. So it's public "just in case" plugin writer decides to implement his callback without extending JBossProductDiscoveryCallback and still wants to set resource name/description the general way. Do you think this use case is not likely? I am not sure